src/commands/watch.rs

```diff
@@ -304,203 +309,35 @@ impl WatchCommand {
-            .unwrap()
-            .glob
-            .ends_with("static-fileserver/spin_static_fs.wasm"));
-        assert!(config
-            .artifact_patterns
-            .get(1)
-            .unwrap()
-            .glob
-            .ends_with("static-fileserver/assets/**/*"));
-        assert!(config
-            .artifact_patterns
-            .get(2)
-            .unwrap()
-            .glob
-            .ends_with("static-fileserver/assets2/**/*"));
-    }
+fn path_of_event(event: &watchexec::event::Event) -> Option<&Path> {
+    event.tags.iter().filter_map(path_of_tag).next()
+}
 
-    #[tokio::test]
-    async fn test_accepts_directory() {
-        let app_path = "tests/watch/static-fileserver";
-        let watch_command = WatchCommand {
-            app_source: app_path.into(),
-            clear: false,
-            debounce: 100,
-            skip_build: false,
-            up_args: vec![],
-        };
-        let config = watch_command.generate_filter_config().await.unwrap();
-
-        assert_eq!(config.source_patterns.len(), 0);
-
-        assert_eq!(config.artifact_patterns.len(), 3);
-        assert!(config
-            .artifact_patterns
-            .get(0)
-            .unwrap()
-            .glob
-            .ends_with("static-fileserver/spin_static_fs.wasm"));
-        assert!(config
-            .artifact_patterns
-            .get(1)
-            .unwrap()
-            .glob
-            .ends_with("static-fileserver/assets/**/*"));
-        assert!(config
-            .artifact_patterns
-            .get(2)
-            .unwrap()
-            .glob
-            .ends_with("static-fileserver/assets2/**/*"));
+fn path_of_tag(tag: &watchexec::event::Tag) -> Option<&Path> {
+    match tag {
+        watchexec::event::Tag::Path { path, .. } => Some(path),
+        _ => None,
     }
+}
 
-    #[tokio::test]
-    async fn test_skip_build_config_proj2() {
-        let app_path = "tests/watch/static-fileserver/spin.toml";
-        let watch_command = WatchCommand {
-            app_source: app_path.into(),
-            clear: false,
-            debounce: 100,
-            skip_build: true,
-            up_args: vec![],
-        };
-        let config = watch_command.generate_filter_config().await.unwrap();
-
-        assert_eq!(config.source_patterns.len(), 0);
-
-        assert_eq!(config.artifact_patterns.len(), 3);
-        assert!(config
-            .artifact_patterns
-            .get(0)
-            .unwrap()
-            .glob
-            .ends_with("static-fileserver/spin_static_fs.wasm"));
-        assert!(config
-            .artifact_patterns
-            .get(1)
-            .unwrap()
-            .glob
-            .ends_with("static-fileserver/assets/**/*"));
-        assert!(config
-            .artifact_patterns
-            .get(2)
-            .unwrap()
-            .glob
-            .ends_with("static-fileserver/assets2/**/*"));
+fn stringise_path(path: &Path) -> anyhow::Result<String> {
```

Comment:
Mega nit: I know we're in the land of made up terms here but this is how I would spell it.
```suggestion
fn stringize_path(path: &Path) -> anyhow::Result<String> {
```

Reply:
oops, thanks